.github/workflows/release.yml

```diff
@@ -0,0 +1,110 @@
+name: Release Creation
+
+on:
+  push:
+    tags:
+      - 'test-*'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+
+    # Load the system.json manifest into memory
+    - name: Load system manifest
+      id: manifest
+      uses: zoexx/github-action-json-file-properties@release
+      with:
+        file_path: "./system.json"
+
+    # Set up variables for future use
+    - name: Set up variables
+      id: get_vars
+      run: |
+        TAG=${GITHUB_REF/refs\/tags\//}
+        echo "TAG_NAME=$TAG" >> $GITHUB_ENV
+        echo "ZIP_NAME=lancer-$TAG.zip" >> $GITHUB_ENV
+        echo "RELEASE_DOWNLOAD_URL=https://github.com/${{github.repository}}/releases/download/$TAG/lancer-$TAG.zip" >> $GITHUB_ENV
+        echo "RELEASE_INSTALL_URL=https://github.com/${{github.repository}}/releases/download/$TAG/system.json" >> $GITHUB_ENV
+
+    # Verify correct naming
+    - name: Verify correct naming
+      env:
+        TAG_NAME: ${{ env.TAG_NAME }}
+        RELEASE_DOWNLOAD: ${{ env.RELEASE_DOWNLOAD_URL }}
+        PACKAGE_VERSION: ${{ steps.manifest.outputs.version }}
+        PACKAGE_DOWNLOAD: ${{ steps.manifest.outputs.download }}
+      run: |
+        if [[ ! $TAG_NAME == release-$PACKAGE_VERSION ]]; then
+          echo "The system.json version does not match tag name."
+          echo "system.json: $PACKAGE_VERSION"
+          echo "tag name: $TAG_NAME"
+          echo "Please fix this and push the tag again."
+          exit 1
```

Comment:
Ah, this should already bail if `system.json` and the tag name don't match, great!